core/src/marker/markerManager.cpp

```diff
@@ -335,13 +336,35 @@ bool MarkerManager::buildGeometry(Marker& marker, int zoom) {
 
     styler->setup(marker, zoom);
 
+    uint32_t selectionColor = 0;
+    bool interactive = false;
+    if (rule->get(StyleParamKey::interactive, interactive) && interactive) {
+        if (selectionColor == 0) {
+            selectionColor = m_scene->featureSelection()->nextColorIdentifier();
+        }
+        rule->selectionColor = selectionColor;
+    } else {
+        rule->selectionColor = 0;
+    }
+
     if (!styler->addFeature(*feature, *rule)) { return false; }
 
+    marker.setSelectionColor(selectionColor);
     marker.setMesh(styler->style().getID(), zoom, styler->build());
 
     return true;
 }
 
+const Marker* MarkerManager::getMarkerOrNullBySelectionColor(uint32_t selectionColor) const {
+    for (const auto& marker : m_markers) {
+        if (marker->selectionColor() == selectionColor) {
```

Comment:
Why is it selection **color**? I'd expect the id of a marker to be of relevance for selection, not the color.